runelite-client/src/main/java/net/runelite/client/plugins/statusbars/BarRenderer.java

```diff
@@ -0,0 +1,170 @@
+		currentValue = currentValueSupplier.get();
+	}
+
+	private void renderBar(Graphics2D graphics, int x, int y, int height)
+	{
+		final Color fill = colorSupplier.get();
+
+		graphics.setColor(BACKGROUND);
+		graphics.drawRect(x, y, WIDTH - PADDING, height - PADDING);
+		graphics.fillRect(x, y, WIDTH, height);
+
+		final int filledHeight = getBarHeight(maxValue, currentValue, height);
+		graphics.setColor(fill);
+		graphics.fillRect(x + PADDING,
+				y + PADDING + (height - filledHeight),
+				WIDTH - PADDING * OFFSET,
+				filledHeight - PADDING * OFFSET);
+	}
+
+	private void renderIconsAndCounters(Graphics2D graphics, int x, int y)
+	{
+		final String counterText = Integer.toString(currentValue);
+		final int widthOfCounter = graphics.getFontMetrics().stringWidth(counterText);
+		final int centerText = (WIDTH - PADDING) / 2 - (widthOfCounter / 2);
+
+		if (config.enableCounter())
+		{
+			graphics.setFont(FontManager.getRunescapeSmallFont());
+			textComponent.setColor(Color.WHITE);
+			textComponent.setText(counterText);
+			textComponent.setPosition(new java.awt.Point(x + centerText + 1, y + COUNTER_ICON_HEIGHT));
+		}
+		else
+		{
+			textComponent.setText("");
+		}
+
+		if (config.enableSkillIcon())
+		{
+			final Image icon = iconSupplier.get();
+			graphics.drawImage(icon, x + ICON_AND_COUNTER_OFFSET_X + PADDING, y + ICON_AND_COUNTER_OFFSET_Y - icon.getWidth(null), null);
+			textComponent.setPosition(new java.awt.Point(x + centerText + 1, y + SKILL_ICON_HEIGHT));
+		}
+
+		textComponent.render(graphics);
+	}
+
+	private void renderRestore(Graphics2D graphics, int x, int y, int height)
+	{
+		final Color color = healColorSupplier.get();
+		final int heal = healSupplier.get();
+
+		if (heal <= 0)
+		{
+			return;
+		}
+
+		final int filledCurrentHeight = getBarHeight(maxValue, currentValue, height);
+		int filledHeight = getBarHeight(maxValue, heal, height);
+		graphics.setColor(color);
+
+		if (filledHeight + filledCurrentHeight > height)
+		{
+			final int overHeal = filledHeight + filledCurrentHeight - height;
+			filledHeight = filledHeight - overHeal + OVERHEAL_OFFSET;
+			graphics.setColor(OVERHEAL_COLOR);
+			graphics.fillRect(x + PADDING,
+					y - filledCurrentHeight + (height - filledHeight) + HEAL_OFFSET,
+					WIDTH - PADDING * OVERHEAL_OFFSET,
+					filledHeight - PADDING * OVERHEAL_OFFSET);
+		}
+		else
+		{
+			graphics.fillRect(x + PADDING,
+					y - OVERHEAL_OFFSET - filledCurrentHeight + (height - filledHeight) + HEAL_OFFSET,
+					WIDTH - PADDING * OVERHEAL_OFFSET,
+					filledHeight + OVERHEAL_OFFSET - PADDING * OVERHEAL_OFFSET);
+		}
+	}
+
+	private static int getBarHeight(int base, int current, int size)
+	{
+		final double ratio = (double) current / base;
+
+		if (ratio >= 1)
+		{
+			return size;
+		}
+
+		return (int) Math.round(ratio * size);
+	}
+
+	public void draw(Graphics2D graphics, int x, int y, int height)
+	{
+		refreshSkills();
+		renderBar(graphics, x, y, height);
+		if (config.enableRestorationBars())
```

Comment:
Both of these config checks seem like they should be done within `renderBar()`, as they're effectively tied to the bar rendering itself.

Also, we generally discourage brace-less statements; change this to be in this format:

```
if (config.enableRestorationBars())
{
    renderRestore(graphics, x, y, height);
}
```

Reply:
Also, I'd be tempted to move these config checks to within these methods since the passed parameters don't depend on the config values.